docs/docs/guides/wagmi_usage/wagmi.md

```diff
@@ -0,0 +1,51 @@
+---
+sidebar_position: 1
+sidebar_label: 'Wagmi Web3js Adaptor'
+title: 'Wagmi Web3js Adaptor'
+---
+
+If you're using [Wagmi](https://wagmi.sh/react/getting-started#use-wagmi) and want to add web3.js, use this provider in your project:
+
+
+```typescript
+import {Web3} from 'web3'
+import {useMemo} from 'react'
+import type {Chain, Client, Transport} from 'viem'
+import {type Config, useClient} from 'wagmi'
+
+export function clientToWeb3js(client?: Client<Transport, Chain>) {
+    if (!client) {
+        return new Web3()
+    }
+
+    const {transport} = client
+
+    if (transport.type === 'fallback') {
+        return new Web3(transport.transports[0].value.url)
+    }
+
+    return new Web3(transport.url)
+}
+
+/** Action to convert a viem Client to a web3.js Instance. */
+export function useWeb3js({chainId}: { chainId?: number } = {}) {
```

Comment:
@avkos are these manually tested ?

Reply:
yes. here is the [dapp example ](https://github.com/avkos/wagmi-web3js-example-app)